UofS-Pulse-Binfo/analyzedphenotypes
# 7.x 3.x - Analyzed Phenotypes Data Downloader

Analyzed Phenotypes Data Downloader: allows user to download a subset or the full set of analyzed phenotypic data uploaded using the Analyzed Phenotypes Data Loader/Uploader. This module features easy selection of option per filter, AJAX add and remove field used to indicate specific germplasm name as a filter, custom column picker/manager (to include and exclude column header) with previewer and support for xlsx, tsv and csv export file types.

**TO Test**
NOTE: This module requires Analyzed Phenotype Data Loader or Uploader to populate the system with sample analyzed phenotypic data. I have test files 1. all 324 germplasm in LDP with one trait, 2. 17 Trait for 100 germplasm in LDP and 3. a file using same location, trait, year and germplasm to test average replicate option - Let me know so I can forward these files to you and save you from a lot of typing. Analyzed Phenotypes Standard Spreadsheet is available in the AP Uploader Branch/pull request.

1. When with data, on page load the form will show a set of experiments available in the system.
The experiment field in "Select Trait" fieldset is the only field enabled to indicate an experiment or experiments is required for other filters/options to activate. Check or click desired experiment(s) to select or hover the mouse pointer to an entry to view a quick data count summary.

2. When choosing a single project, a field proceeding experiment field becomes available and selecting an option in that field activates the next field and so on. Providing an option to all fields in "Select Trait" Fieldset would be sufficient to export data. Choosing multiple projects will auto select all relevant options based on the assumption that user wants all data from both/all projects selected.

3. More optional filters available in "Additional filter criteria (Optional)".

4. As you select filters, the system fetches all germplasm names used, based on all options provided so far. A summary count of germplasm found is provided and a view window lists each item. An AJAX field underneath is provided to further filter the result set to only germplasm names entered. Add and Delete buttons next to each field are available to managed names entered.

5. Export file type, average replicate and R options are available in the last fieldset "Choose your output file". Similar to germplasm names, as you select options, the system gathers pertinent column headers (default) required to generate a file. Optional headers are sometimes available in case additional but optional information is required and these headers can be selected to include in the default set, by checking a corresponding check box. To preview headers selected with a sample data, "Click preview" headers link.

6. When complete, Click Download button.

@laceysanderson I think I need to walk you through the AP Uploader, Downloader, Visulizer and everything in between if you have time. Thanks!
